src/internal/pfssync/sync.go

```diff
@@ -2,17 +2,133 @@ package pfssync
 
 import (
 	"archive/tar"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"syscall"
 
 	"github.com/pachyderm/pachyderm/v2/src/client"
 	"github.com/pachyderm/pachyderm/v2/src/internal/tarutil"
 	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"golang.org/x/sync/errgroup"
 )
 
-// Pull pulls a file from PFS and stores it in the local filesystem.
-func Pull(pachClient *client.APIClient, file *pfs.File, storageRoot string, cb ...func(*tar.Header) error) error {
-	r, err := pachClient.GetTarFile(file.Commit.Repo.Name, file.Commit.ID, file.Path)
+// Importer is the standard interface for a PFS importer.
+type Importer interface {
```

Comment:
The naming seems kind of weird, I don't generally think of downloading data to disk as "importing" I remember about naming this though and didn't find anything that seemed right. But maybe just `Syncer` since it's the sync package?

Reply:
Yeah, I was thinking more about going from the PFS to local filesystem format, but it probably makes more sense to emphasize the downloading aspect in the name. I don't think `Syncer` is a good name because we are not syncronizing data with PFS through this particular abstraction (that does sort of draw into question the name of the package though), we are just downloading / pulling. Should I just change this back to Puller or does Downloader seem preferable? Or something else?

Reply:
Downloader seems great actually, that's exactly what it's doing.